src/thor/triplegbuilder.cc

```diff
@@ -102,7 +119,9 @@ void AssignAdmins(const AttributesController& controller,
 void UpdateIncident(const std::shared_ptr<const valhalla::IncidentsTile>& incidents_tile,
                     TripLeg& leg,
                     const valhalla::IncidentsTile::Location* incident_location,
-                    uint32_t index) {
+                    uint32_t index,
+                    const graph_tile_ptr& tile,
+                    const valhalla::baldr::DirectedEdge* de) {
```

Comment:
This is the opposing edge of the edge on which the incident is present. I'm using it to get the start node of the incident edge.

Reply:
I don't quite understand this comment. Why is this the opposing edge?

Reply:
Consider the following case:
```
         i   i
... A----+---+--->B ...
```
Incident is between the 2 `i`'s. Node A is in country C1 & B in country C2. In triplegbuilder when consuming edge A->B the incident needs to be associated with the country code of A (since we haven't crossed the border into country C2 of B, CMIIW here). Therefore, we get the opposing edge B->A and use its end node (which is A) to get the country code (since there's no API to get the start node given a directed edge).

The whole premise of doing this is that an incident needs to be associated with the country of the node just prior to the start of the incident.

Reply:
I see. Could the variable be named `opposite_edge` instead of `de` though?

Reply:
in a lot of places we use the end node of the edge to denote which admin its in since we dont split the edges where admin boundaries occur begin node is as good as end node. i would suggest simplifying by just using the end node of the edge that the incident is on as its simpler to get and will be just as in correct in the extreme edge cases where it would occur